src/NumericalAlgorithms/Interpolation/LinearLeastSquares.cpp

```diff
@@ -70,6 +70,22 @@ std::array<double, Order + 1> LinearLeastSquares<Order>::fit_coefficients(
   return coefficients;
 }
 
+template <size_t Order>
+template <typename T>
+std::vector<std::array<double, Order + 1>>
+LinearLeastSquares<Order>::fit_coefficients(const T& x_values,
+                                            const std::vector<T>& y_values) {
+  ASSERT(not y_values.empty(), "Must have non-zero series");
+  std::vector<std::array<double, Order + 1>> fit_coeffs{};
+  ASSERT(x_values.size() == y_values[0].size(),
```

Comment:
I'd say put this inside the loop and do `y_values[curve_index]` since each one could be a different length in theory